.travis.yml

```diff
@@ -7,117 +7,131 @@ env:
 
 before_script:
   - echo -e "Host github.com\n\tStrictHostKeyChecking no\n" >> ~/.ssh/config
-  - if [[ $DEVITO_BACKEND == 'yask' ]]; then
+  - if [[ ${DEVITO_BACKEND} == 'yask' ]]; then
       conda install swig; cd ../;
       git clone https://github.com/opesci/yask.git;
       cd yask; make compiler && make compiler-api; pip install -e .; cd ../devito;
     fi
 
 script:
   - flake8 --builtins=ArgumentError .
-  # Run tests with pip
-  - if [[ $INSTALL_TYPE == 'pip_setup' ]]; then python setup.py test; fi
-  # Run test with py.tests
-  - if [[ $INSTALL_TYPE == 'conda' ]]; then py.test --cov devito tests/; fi
+  # Run tests
+  - if [[ ${TEST_BUNDLE} == 'unit' && ${INSTALL_TYPE} == 'pip' ]]; then python setup.py test; fi
+  - if [[ ${TEST_BUNDLE} == 'unit' && ${INSTALL_TYPE} == 'conda' ]]; then py.test --cov devito tests; fi
   # Run remaining specialized examples and tests
   # Additional seismic operator tests
-  - if [[ $RUN_EXAMPLES == 'True' ]]; then python examples/seismic/benchmark.py test -P tti -so 4 -a -d 20 20 20 -n 5; fi
-  - if [[ $RUN_EXAMPLES == 'True' ]]; then python examples/seismic/benchmark.py test -P acoustic -a; fi
-  - if [[ $RUN_EXAMPLES == 'True' ]]; then python examples/seismic/acoustic/acoustic_example.py --full; fi
-  - if [[ $RUN_EXAMPLES == 'True' ]]; then python examples/seismic/acoustic/acoustic_example.py --full --checkpointing; fi
-  - if [[ $RUN_EXAMPLES == 'True' ]]; then python examples/seismic/acoustic/acoustic_example.py --constant --full; fi
-  - if [[ $RUN_EXAMPLES == 'True' ]]; then python examples/misc/linalg.py mat-vec mat-mat-sum transpose-mat-vec; fi
-  - if [[ $RUN_EXAMPLES == 'True' ]]; then python examples/seismic/tti/tti_example.py -a; fi
-  - if [[ $RUN_EXAMPLES == 'True' ]]; then python examples/seismic/tti/tti_example.py -a --noazimuth; fi
-  # Test tutorial notebooks for the website using nbval
-  - if [[ $RUN_EXAMPLES == 'True' ]]; then py.test --nbval examples/cfd; fi
-  - if [[ $RUN_EXAMPLES == 'True' ]]; then py.test --nbval examples/seismic/tutorials/0[1-3]*; fi
+  - if [[ ${TEST_BUNDLE} == 'tutorials' ]]; then python examples/seismic/benchmark.py test -P tti -so 4 -a -d 20 20 20 -n 5; fi
+  - if [[ ${TEST_BUNDLE} == 'tutorials' ]]; then python examples/seismic/benchmark.py test -P acoustic -a; fi
+  - if [[ ${TEST_BUNDLE} == 'tutorials' ]]; then python examples/seismic/acoustic/acoustic_example.py --full; fi
+  - if [[ ${TEST_BUNDLE} == 'tutorials' ]]; then python examples/seismic/acoustic/acoustic_example.py --constant --full; fi
+  - if [[ ${TEST_BUNDLE} == 'tutorials' ]]; then python examples/misc/linalg.py mat-vec mat-mat-sum transpose-mat-vec; fi
+  - if [[ ${TEST_BUNDLE} == 'tutorials' ]]; then python examples/seismic/tti/tti_example.py -a; fi
+  - if [[ ${TEST_BUNDLE} == 'tutorials' ]]; then py.test --nbval examples/cfd; fi
+  - if [[ ${TEST_BUNDLE} == 'tutorials' ]]; then py.test --nbval examples/seismic/tutorials; fi
```

Comment:
Some of the previous tests are now missing:

- python examples/seismic/acoustic/acoustic_example.py --full --checkpointing
- python examples/seismic/tti/tti_example.py -a --noazimuth